src/lsp/cobol_typeck/typeck_procedure.ml

```diff
@@ -212,10 +212,19 @@ let references ~(data_definitions: Cobol_unit.Types.data_definitions) procedure
           error acc @@ Ambiguous_data_name { given_qualname = qn &@ loc;
                                              matching_qualnames }
 
-    method! fold_procedure_section s ({ current_section; _ } as acc) =
-      Visitor.do_children_and_then
-        { acc with current_section = Some s }
-        (fun acc -> { acc with current_section })
+    method! fold_procedure_section ({ section_paragraphs; _ } as s)
+        ({ current_section; _ } as acc) =
+      { acc with current_section = Some s }
+      |> Cobol_common.Visitor.fold_list v section_paragraphs.list
+        ~fold:Cobol_unit.Visitor.fold_procedure_paragraph'
+      |> (fun acc -> { acc with current_section })
+      |> Cobol_common.Visitor.skip
```

Comment:
```suggestion
      let acc =
        Visitor.fold_list v section_paragraphs.list
          ~fold:Cobol_unit.Visitor.fold_procedure_paragraph'
          { acc with current_section = Some s }
      in
      Visitor.skip { acc with current_section }
```
Please use the reverse style (valid comment in general): `|>` is nice in some places, but doing it the other way round with `@@` in this context helps easily detect what the main behavior is (whether it's a `skip`, or something else). Or like in the suggestion.